sdk/appconfiguration/azure-data-appconfiguration/src/configuration_client.cpp

```diff
@@ -0,0 +1,2316 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) TypeSpec Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+#include "azure/data/appconfiguration/configuration_client.hpp"
+
+#include "private/package_version.hpp"
+
+#include <azure/core/exception.hpp>
+#include <azure/core/http/http.hpp>
+#include <azure/core/http/http_status_code.hpp>
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/internal/json/json.hpp>
+#include <azure/core/io/body_stream.hpp>
+
+#include <functional>
+
+using namespace Azure::Data::AppConfiguration;
+using namespace Azure::Data::AppConfiguration::_detail;
+
+GetKeysResponseContentType const
+    GetKeysResponseContentType::ApplicationVndMicrosoftAppconfigKeysetJson{
+        "application/vnd.microsoft.appconfig.keyset+json"};
+GetKeysResponseContentType const GetKeysResponseContentType::ApplicationProblemJson{
+    "application/problem+json"};
+
+GetKeyValuesResponseContentType const
+    GetKeyValuesResponseContentType::ApplicationVndMicrosoftAppconfigKvsetJson{
+        "application/vnd.microsoft.appconfig.kvset+json"};
+GetKeyValuesResponseContentType const GetKeyValuesResponseContentType::ApplicationProblemJson{
+    "application/problem+json"};
+
+KeyValueFields const KeyValueFields::Key{"key"};
+KeyValueFields const KeyValueFields::Label{"label"};
+KeyValueFields const KeyValueFields::ContentType{"content_type"};
+KeyValueFields const KeyValueFields::Value{"value"};
+KeyValueFields const KeyValueFields::LastModified{"last_modified"};
+KeyValueFields const KeyValueFields::Tags{"tags"};
+KeyValueFields const KeyValueFields::Locked{"locked"};
+KeyValueFields const KeyValueFields::Etag{"etag"};
+
+PutKeyValueRequestContentType const
+    PutKeyValueRequestContentType::ApplicationVndMicrosoftAppconfigKvJson{
+        "application/vnd.microsoft.appconfig.kv+json"};
+PutKeyValueRequestContentType const
+    PutKeyValueRequestContentType::ApplicationVndMicrosoftAppconfigKvsetJson{
+        "application/vnd.microsoft.appconfig.kvset+json"};
+PutKeyValueRequestContentType const PutKeyValueRequestContentType::ApplicationJson{
+    "application/json"};
```

Comment:
https://github.com/Azure/autorest.cpp/issues/426

Given how services have varying values, it might not be worth the complexity to implement the emitter in a way to track the existing values present in `HttpShared` and only use it for such content-types, while using custom ones for others.